indigo_api/templatetags/indigo.py

```diff
@@ -19,3 +19,49 @@ def work_resolver_url(context, work):
         frbr_uri = '/akn' + frbr_uri
 
     return context.get('resolver_url', settings.RESOLVER_URL) + frbr_uri
+
+
+@register.simple_tag
+def commenced_provisions_description(document, commencement, uncommenced=False):
+    work = document.work
+    if uncommenced:
+        uncommenced_provisions = document.work.uncommenced_provisions()
+        # TODO: same as below for uncommenced provisions
+        return ', '.join([p.id for p in uncommenced_provisions])
+
+    commenceable_provisions = work.commenceable_provisions()
+    commenced_provisions = commencement.provisions
+    runs = []
```

Comment:
If you move the stuff below this into a helper function, passing in the various provision lists, then your TODO just above should be a matter of pulling out uncommenced provisions and calling the helper function.